webguitest/fabi/js/flipcomm.js

```diff
@@ -0,0 +1,527 @@
+        thiz.pauseLiveValueListener();
+        increaseProgress(10);
+        var percentPerSlot = 50 / thiz.getSlots().length;
+        var saveSlotsPromise = new Promise(function (resolve) {
+            loadAndSaveSlot(thiz.getSlots(), 0);
+
+            function loadAndSaveSlot(slots, i) {
+                if (i >= slots.length) {
+                    resolve();
+                    return;
+                }
+                var slot = slots[i];
+                thiz.testConnection().then(function () {
+                    loadSlotByConfig(slot);
+                    thiz.testConnection().then(function () {
+                        sendAtCmdNoResultHandling('AT SA ' + slot);
+                        increaseProgress(percentPerSlot);
+                        loadAndSaveSlot(slots, i + 1);
+                    });
+                });
+            }
+        });
+
+        function increaseProgress(percent) {
+            progress += percent;
+            updateProgressHandler(progress);
+        }
+
+        return new Promise(function (resolve) {
+            saveSlotsPromise.then(function () {
+                thiz.testConnection().then(function () {
+                    increaseProgress(30);
+                    loadSlotByConfig(_currentSlot).then(function () {
+                        increaseProgress(10);
+                        thiz.resumeLiveValueListener();
+                        resolve();
+                    });
+                });
+            });
+        });
+    };
+
+    thiz.calibrate = function () {
+        sendAtCmdNoResultHandling('AT CA');
+        return thiz.testConnection();
+    };
+
+    thiz.rotate = function () {
+        var currentOrientation = thiz.getConfig(thiz.ORIENTATION_ANGLE);
+        thiz.setValue(thiz.ORIENTATION_ANGLE, (currentOrientation + 90) % 360, 0);
+        sendAtCmdNoResultHandling('AT CA');
+        return thiz.testConnection();
+    };
+
+    thiz.startLiveValueListener = function (handler) {
+        console.log('starting listening to live values...');
+        setLiveValueHandler(handler);
+    };
+
+    thiz.stopLiveValueListener = function () {
+        setLiveValueHandler(null);
+        console.log('listening to live values stopped.');
+    };
+
+    thiz.pauseLiveValueListener = function () {
+        thiz.sendATCmd('AT ER');
+        console.log('listening to live values stopped.');
+    };
+
+    thiz.resumeLiveValueListener = function () {
+        if (_valueHandler) {
+            thiz.sendATCmd('AT SR');
+            console.log('listening to live values resumed.');
+        } else {
+            console.log('listening to live values not resumed, because no value handler.');
+        }
+    };
+
+    // return 'true' if SipAndPuff device is available, else 'false'
+    thiz.isSipAndPuffLive = function () {
+        return new Promise(
+            function(resolve, reject) {
+		thiz.startLiveValueListener(function (livedata) {
+		    if (livedata) {
+			resolve(true);
```

Comment:
the problem here is that the listener is not closed after the check. So please add a thiz.stopLiveValueListener() at line 257. Additionally I would store the result of this check to a local variable and if it was `true` before, this method should return `true` again without really opening the connection to the FABI again.

Reply:
Okay. Let me try it.